src/Microsoft.AspNet.Mvc.DataAnnotations/DataAnnotationsModelValidator.cs

```diff
@@ -38,6 +148,8 @@ public IEnumerable<ModelValidationResult> Validate(ModelValidationContext valida
                 MemberName = memberName
             };
 
+            var metadataProvider = (IModelMetadataProvider)context.GetService(typeof(IModelMetadataProvider));
```

Comment:
Instead add an `IModelMetadataProvider` parameter to the `DataAnnotationsModelValidatorProvider` constructor and pass the value into the `DataAnnotationsModelValidator` constructor.

FYI not relevant anymore but it's almost never correct to call `GetService()`. Plus we have convenience helpers like `GetRequiredService<TService>()`.